content/client-lib-development-guide/features.textile

```diff
@@ -587,6 +587,7 @@ h3(#eventemitter). EventEmitter mixin / interface
 * @(RTE4)@ @EventEmitter#once@ registers the provided listener for either the first event that is emitted when no @event@ argument is provided, or for only the first occurrence of a single named event when an @event@ argument is provided. If @once@ is called more than once with the same listener, the listener is added multiple times to its listener registry. Therefore, as an example, assuming the same listener is registered twice using @once@, and an event is emitted once, the listener would be invoked twice. However, all subsequent events emitted would not invoke the listener as @once@ ensures that each registration is only invoked once
 * @(RTE5)@ @EventEmitter#off@ deregisters a listener. If called with a specific event and a listener, it removes all registrations that match both the given listener and the given event; if called only with a listener, it removes all registrations matching the given listener, regardless of whether they are associated with an event or not; if called with no arguments, it removes all registrations, for all events and listeners
 * @(RTE6)@ @EventEmitter#emit@ emits an event, calling registered listeners with the given event name and any other given arguments. If an exception is raised in any of the listeners, the exception is caught by the @EventEmitter@ and the exception is logged to the Ably logger. Tests must exist to ensure exceptions raised in client code do not propagate and inhibit other event processing within the client library
+** @(RTE6b)@ Every listener that is registered at the time @emit@ is invoked should be called exactly once. A test should assert that any listener registered and/or removed within an already existing @on@ listener does not result in the newly added listeners being called or the newly removed listeners not being called when an event is emitted, see "https://goo.gl/OVTtjO":https://goo.gl/OVTtjO
```

Comment:
> RTE6b

Why not RTE6a?

> `Every listener that is registered at the time @emit@ is invoked should be called exactly once.`

This seems like an odd way of putting the requirement. E.g. if I add an onMessage listener within an onMessage callback, I expect it to be called maybe hundreds of times, whenever a message arrives, not 'exactly once'. I just don't want it to be called on the current invocation of `emit`.

Alternative wording proposal: https://github.com/ably/docs/compare/source-0-10...event-emitter-emit-spec-update-alternative?expand=1